### PR TITLE
[Authentication] Route users to error page when Auth service is unreachable

### LINKED ISSUE #693: [Login] Route users to Error page when Auth service unreachable
Image

### PR DESCRIPTION
**Pull request by meissadia**
Close #693 

## Changes

- Remove "Pre-MVP" from 500 page `Error details`
- Route users to 500 page when Authentication service is unreachable

## Considerations
- Is this a good place for the redirect? I needed a place inside the BrowserRouter (to support use of Navigate component) but didn't want to stick it in a component that was too specific so that this logic would apply to *any* authentication error.
- Should this be encapsulated in a custom hook so that we can apply this logic in a more targeted way?
- Can we simply label the `Error details` expandable as `Error details`? Does it *really* need qualifying (i.e. "for Pre-MVP..." or "for Beta")?

## How to test this PR

1. Stop Keycloak container
2. Visit unauthenticated homepage
3. Try to log-in
4. Observe that you are redirected to the `/500` page with `Error details` displayed

## Screenshots
https://github.com/cfpb/sbl-frontend/assets/2592907/26c5dc85-c00b-4088-a732-0165a81528bb

### CODE REVIEW

**Review by billhimmelsbach — CHANGES_REQUESTED** [review]
Heyo! Looks good: I do think we'll want to look into retries in a later PR, and maybe refine this a bit with a loading indicator on the button. Would you mind making a ticket for that?

For now, just a small text change. 👍

File: src/pages/Error/Error500.tsx [hunk 1]
```diff
@@ -38,13 +36,13 @@ function ErrorDetails(): ReactElement | null {
...
-  if (!message) return null;
+  const displayText = [code, message].filter(Boolean).join(' - ');
 
-  const displayText = `${code ?? DEFAULT_STATUS} - ${message}`;
+  if (displayText.length === 0) return null;
 
   return (
     <div className='error-details-wrapper mt-[30px] w-full'>
-      <Expandable header='Error details for pre-MVP developers'>
+      <Expandable header='Error details for SBL Devs'>
```
[the remark this answers is not in the document]
Comment (meissadia) [reply src/pages/Error/Error500.tsx:45]: @billhimmelsbach Can we just simplify and trim to it "Error details"? It doesn't seem like the type of thing that requires qualification.  
 
I don't recall, but Is this details element going away after Production launch? Seems like a useful element regardless of which phase we're in.

[the remark this answers is not in the document]
Comment (billhimmelsbach) [reply src/pages/Error/Error500.tsx:45]: Sounds good @meissadia. It's possible there's weird stuff in there, or they might send an error that had their own financial institution info in it, but I think it's ok just to make it "error details" for now

**Review by billhimmelsbach — COMMENTED** [review]
> @billhimmelsbach Can we just simplify and trim to it "Error details"? It doesn't seem like the type of thing that requires qualification.

This sounds good, make the change and I'll approve it.👍

**Comment by meissadia** [comment]
> > @billhimmelsbach Can we just simplify and trim to it "Error details"? It doesn't seem like the type of thing that requires qualification.
> 
> This sounds good, make the change and I'll approve it.👍

@billhimmelsbach Updated
- https://github.com/cfpb/sbl-frontend/issues/731
- https://github.com/cfpb/sbl-frontend/issues/732

**Review by billhimmelsbach — APPROVED** [review]
Looks good!

**Pull request merged**